android/src/main/java/org/mozilla/mozstumbler/client/Updater.java

```diff
@@ -40,6 +42,12 @@ public boolean wifiExclusiveAndUnavailable(Context c) {
         return !NetworkInfo.getInstance().isWifiAvailable() && ClientPrefs.getInstance(c).getUseWifiOnly();
     }
 
+    public boolean checkForUpdatesRateLimited(final Activity activity, String api_key) {
+        if (System.currentTimeMillis() - sLastUpdateCheck < UPDATE_CHECK_FREQ) {
```

Comment:
The dependency on System.currentTimeMillis() should be replaced with grabbing ISystemClock from the ServiceLocator.

You can then instrument the time by using the MockSystemClock.

https://github.com/mozilla/MozStumbler/blob/dev/android/src/main/java/org/mozilla/mozstumbler/svclocator/services/MockSystemClock.java

```java
        clock = new MockSystemClock();
        ServiceLocator.getInstance().putService(ISystemClock.class, clock);
```

will replace the real clock with something you can play with under test.  The LocationChangeSensorTest is a reasonable example to look at:

https://github.com/mozilla/MozStumbler/blob/3484486a00cd72c07d806b70bb88332f8cfb8c34/android/src/test/java/org/mozilla/mozstumbler/service/stumblerthread/motiondetection/LocationChangeSensorTest.java#L62